core/internal/cltest/cltest.go

```diff
@@ -890,14 +890,14 @@ func WaitForPipelineComplete(t testing.TB, nodeID int, jobID int32, expectedPipe
 		assert.NoError(t, err)
 		var completed []pipeline.Run
 
-		for i := range prs {
-			if prs[i].State == pipeline.RunStatusCompleted {
-				if !prs[i].Outputs.Null {
-					if !prs[i].Errors.HasError() {
+		for _, pr := range prs {
+			if pr.State == pipeline.RunStatusCompleted {
+				if !pr.Outputs.Null {
+					if !pr.Errors.HasError() {
```

Comment:
These two checks can be removed, `RunStatusCompleted` already ensures it's a completed run (& no errors). (The len(taskRuns) check is still required though). I forgot to simplify this when I added `run.State`.

Reply:
Done